packages/url_launcher/url_launcher_linux/linux/url_launcher_plugin.cc

```diff
@@ -45,6 +45,20 @@ static gchar* get_url(FlValue* args, GError** error) {
   return g_strdup(fl_value_get_string(url_value));
 }
 
+// Check if URI has launchable file resource
+static gboolean can_launch_uri_with_file_resource(FlUrlLauncherPlugin* self,
+                                                  const gchar* url) {
+  g_autoptr(GError) error = nullptr;
+  g_autoptr(GFile) file = g_file_new_for_uri(url);
+  g_autoptr(GAppInfo) app_info =
+      g_file_query_default_handler(file, NULL, &error);
+  gboolean is_launchable = FALSE;
+  if (app_info != nullptr) {
+    is_launchable = TRUE;
+  }
+  return is_launchable;
```

Comment:
These 5 lines collapses to `return app_info != nullptr;` now that you are using autoptrs